src/screens/SearchScreen.js

```diff
@@ -3,49 +3,40 @@ import {View, Text, ActivityIndicator, StyleSheet, Platform, Keyboard} from 'rea
 import {Button} from 'react-native-elements';
 import {MapView} from 'expo';
 import {Icon} from 'react-native-elements';
+import {connect} from 'react-redux';
+
+import {initializeMap, updateViewStyle, mapLoaded, updateRegion} from "../actions";
 
 import _ from 'lodash';
 
 import {CardSection, Card} from '../components/common/index';
 
-const SearchOptions = {
-    MAP: 'map',
-    LIST: 'list'
-};
+import {map} from '../constants';
 
 
 class SearchScreen extends Component {
-    state = {
-        region: {
-            longitude: -105.727939,
-            latitude: 39.695168,
-            longitudeDelta: 1,
-            latitudeDelta: 1
-        },
-        mapLoaded: false,
-        viewStyle: SearchOptions.MAP
-    };
+    componentWillMount() {
+        this.props.initializeMap();
+    }
+
+    componentDidMount() {
+        Keyboard.dismiss();
+
+        this.props.navigation.setParams({buttonName: map.SearchOptions.LIST, toggleButton: this.toggleButton});
+
+        this.props.mapLoaded();
+    }
 
     toggleButton = () => {
-        const {viewStyle} = this.state;
+        const {viewStyle} = this.props;
 
         const nextButtonName = viewStyle;
-        const newViewStyle = viewStyle === SearchOptions.MAP ? SearchOptions.LIST : SearchOptions.MAP;
+        const newViewStyle = viewStyle === map.SearchOptions.MAP ? map.SearchOptions.LIST : map.SearchOptions.MAP;
 
         this.props.navigation.setParams({buttonName: nextButtonName});
-        this.setState({viewStyle: newViewStyle});
+        this.props.updateViewStyle(newViewStyle);
```

Comment:
Above are several examples of the refactor: calling `props` instead of `state` for attributes, using the new constants file instead of a reference to local constants, and calling actions off `props` to trigger state changes.